crates/agglayer-certificate-orchestrator/src/lib.rs

```diff
@@ -0,0 +1,216 @@
+use std::{
+    collections::{BTreeMap, VecDeque},
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use agglayer_clock::Event;
+use futures_util::{future::BoxFuture, Stream, StreamExt};
+use tokio::{
+    sync::mpsc::Receiver,
+    task::{JoinHandle, JoinSet},
+};
+use tokio_util::sync::{CancellationToken, WaitForCancellationFutureOwned};
+use tracing::{debug, error};
+
+#[cfg(test)]
+mod tests;
+
+const MAX_POLL_READS: usize = 1_000;
+
+/// Certificate orchestrator that receives batches from CDKs.
+/// It collects batches and sends them to the epoch packer when an epoch ends.
+pub struct CertificateOrchestrator<C, A> {
+    /// Epoch packing task resolver.
+    epoch_packing_tasks: JoinSet<Result<(), Error>>,
+    /// Epoch packing task builder.
+    epoch_packing_task_builder: A,
+    /// Clock stream to receive EpochEnded events.
+    clock: C,
+    /// Batches received from CDKs.
+    received_batches: VecDeque<()>,
```

Comment:
```suggestion
    /// Certificates received from CDKs.
    received_certificates: VecDeque<()>,
```

I see several other occurrences on my `git grep -i batch`